pkg/util/spanlogger/spanlogger.go

```diff
@@ -77,6 +77,31 @@ func FromContextWithFallback(ctx context.Context, fallback log.Logger) *SpanLogg
 // also puts the on the spans.
 func (s *SpanLogger) Log(kvps ...interface{}) error {
```

Comment:
Is it doable adding some valuable unit tests on the new logic you've added? Otherwise we'll need to manually test it before merging, cause the logic is a bit convoluted and there may be subtle bugs slipping it.

Reply:
Because I could not see a way to retrieve the field of SpanLogger, it's unlikely to test by comparing the value of the fields. If we refactor the new logic into a private function, I can write unit tests for it. However unit testing a private function doesn't sound like a good idea.

Reply:
@pracucci Do you have any further suggestion based on my previous comment? Thanks!

Reply:
How about using https://github.com/opentracing/opentracing-go/blob/master/mocktracer/mocktracer.go ?

(We unit-test private functions all the time, but I agree it's better if you can avoid)